File: java/src/main/java/com/ccpayment/services/UserWithdrawService.java
package com.ccpayment.services;

import com.ccpayment.APIError;
import com.ccpayment.Client;
import com.google.gson.JsonObject;

import java.io.IOException;

public class UserWithdrawService {
    private final Client client;

    public UserWithdrawService(Client client) {
        this.client = client;
    }

    public JsonObject applyUserWithdrawToNetwork(JsonObject data) throws APIError, IOException {
        return client.post("/applyUserWithdrawToNetwork", data, JsonObject.class);
    }

    public JsonObject applyUserWithdrawToCwallet(JsonObject data) throws APIError, IOException {
        return client.post("/applyUserWithdrawToCwallet", data, JsonObject.class);
    }

    public JsonObject getUserWithdrawRecord(JsonObject data) throws APIError, IOException {
        return client.post("/getUserWithdrawRecord", data, JsonObject.class);
    }

    public JsonObject getUserWithdrawRecordList(JsonObject data) throws APIError, IOException {
        return client.post("/getUserWithdrawRecordList", data, JsonObject.class);
    }
}

File: ruby/lib/ccpayment/services/user_withdraw_service.rb
module CCPayment
  module Services
    class UserWithdrawService
      def initialize(client)
        @client = client
      end

      def apply_user_withdraw_to_network(data)
        @client.post('/applyUserWithdrawToNetwork', data)
      end

      def apply_user_withdraw_to_cwallet(data)
        @client.post('/applyUserWithdrawToCwallet', data)
      end

      def get_user_withdraw_record(data)
        @client.post('/getUserWithdrawRecord', data)
      end

      def get_user_withdraw_record_list(data)
        @client.post('/getUserWithdrawRecordList', data)
      end
    end
  end
end
